code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -516,61 +557,80 @@ GLOBAL_VAR(medibot_unique_id_gen)
 			soft_reset()
 			break
 
-		var/treatment_method = null
-
-		if(C.getBruteLoss() >= heal_threshold)
-			treatment_method = BRUTE
-
-		else if(C.getFireLoss() >= heal_threshold)
-			treatment_method = BURN
 
-		else if(C.getOxyLoss() >= (5 + heal_threshold))
-			treatment_method = OXY
+		var/treat_behaviour = null
+		if(!treat_behaviour && reagent_glass && reagent_glass.reagents.total_volume)
+			for(var/datum/reagent/R in reagent_glass.reagents.reagent_list)
+				if(!C.reagents.has_reagent(R.type) && !check_overdose(patient, R, injection_amount))
+					treat_behaviour = "internal_beaker"
+					break
```

Comment:
I honestly don't understand what you're achieving here

Reply:
I think the code here was frankensteined a bit, new commit should be better